src/main/java/dev/memphis/sdk/ClientOptions.java

```diff
@@ -0,0 +1,104 @@
+package dev.memphis.sdk;
+
+public class ClientOptions {
+    final String host;
+    final String username;
+    final String broker_token;
+    final int port;
+    final boolean reconnect;
+    final int max_reconnect;
+    final int reconnect_interval_ms;
+    final int timeout_ms;
+    final String key_file;
+    final String cert_file;
+    final String ca_file;
+
+    private ClientOptions(Builder b) {
+        this.host = b.host;
+        this.username = b.username;
+        this.broker_token = b.broker_token;
+        this.port = b.port;
+        this.reconnect = b.reconnect;
+        this.max_reconnect = b.max_reconnect;
+        this.reconnect_interval_ms = b.reconnect_interval_ms;
+        this.timeout_ms = b.timeout_ms;
+        this.key_file = b.key_file;
+        this.cert_file = b.cert_file;
+        this.ca_file = b.ca_file;
+    }
+
+    public static class Builder {
+        private String host;
+        private String username;
+        private String broker_token;
+
+        // Connection parameters with default values
+        private int port = 6666;
+        private boolean reconnect = true;
+        private int max_reconnect = 3;
+        private int reconnect_interval_ms = 1500;
+        private int timeout_ms = 1500;
+        private String key_file = "";
+        private String cert_file = "";
+        private String ca_file = "";
+
+        public ClientOptions build() {
```

Comment:
I think it would be good to move some of the validation logic into this method rather than the constructor for the MemphisConnection.